lib/client/keyagent_test.go

```diff
@@ -506,26 +497,22 @@ func (s *KeyAgentTestSuite) makeKey(username string, allowedLogins []string, ttl
 	}, nil
 }
 
-func startDebugAgent() (closer func(), err error) {
-	rand.Seed(time.Now().Unix())
-	socketpath := filepath.Join(os.TempDir(),
-		fmt.Sprintf("teleport-%d-%d.socket", os.Getpid(), rand.Uint32()))
-
+func startDebugAgent(t *testing.T) error {
+	socketpath := filepath.Join(t.TempDir(), "teleport-test")
 	listener, err := net.Listen("unix", socketpath)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return trace.Wrap(err)
 	}
```

Comment:
unix domain socket path is [limited](https://unix.stackexchange.com/questions/367008/why-is-socket-path-length-limited-to-a-hundred-chars) to 104 chars 
thus  the `net.Listen("unix", socketpath)` call will fail for test where name is grater than 32 chars.

Reply:
yes. I saw your change to use `ioutil.TempDir` directly. So I did the same change for #9220 which is the one I am planning to backport. so it will be fixed [there](https://github.com/gravitational/teleport/pull/9220/files#diff-d5297ba22da5138de69b7f33d44d901068991054838ad01371ea3ead74f875e9)

Reply:
Wouldn't it be better to have the fix in this PR as well?